solidity/contracts/vortex/VortexBurner.sol

```diff
@@ -0,0 +1,522 @@
+        }
+
+        // calculate the burn incentive fee and reduce it from the total amount to convert
+        (uint256 netNetworkTokenConversionAmount, uint256 incentiveFeeAmount) =
+            applyIncentiveFee(grossNetworkTokenConversionAmount);
+
+        // convert all network token amounts to the governance token
+        totalGovTokenAmountToBurn = totalGovTokenAmountToBurn.add(
+            network.rateByPath(govPath, netNetworkTokenConversionAmount)
+        );
+        require(totalGovTokenAmountToBurn > 0, "ERR_ZERO_TARGET_AMOUNT");
+
+        return (amounts, networkTokenConversionAmounts, totalGovTokenAmountToBurn, incentiveFeeAmount);
+    }
+
+    /**
+     * @dev transfers the ownership of the network fee wallet
+     *
+     * @param newOwner the new owner of the network fee wallet
+     */
+    function transferNetworkFeeWalletOwnership(address newOwner) external ownerOnly {
+        networkFeeWallet().transferOwnership(newOwner);
+    }
+
+    /**
+     * @dev accepts the ownership of he network fee wallet
+     */
+    function acceptNetworkFeeOwnership() external ownerOnly {
+        networkFeeWallet().acceptOwnership();
+    }
+
+    /**
+     * @dev returns the vortex conversion strategy for the specified tokens
+     *
+     * @param tokens the tokens to convert
+     *
+     * @return the conversion paths for each of tokens to convert
+     * @return the conversion amounts for each of tokens to convert
+     * @return the conversion path for the last network to governance tokens conversion
+     */
+    function vortexStrategy(IERC20[] calldata tokens, ITokenHolder feeWallet)
+        private
+        view
+        returns (
+            address[][] memory,
+            uint256[] memory,
+            address[] memory
+        )
+    {
+        IConverterRegistry registry = converterRegistry();
+
+        // create conversion paths and collect available token amounts
+        address[][] memory paths = new address[][](tokens.length);
+        uint256[] memory amounts = new uint256[](tokens.length);
+
+        for (uint256 i = 0; i < tokens.length; ++i) {
+            IERC20 token = tokens[i];
+
+            address[] memory path = new address[](3);
+            path[0] = address(token);
+
+            // don't look up for a converter for either the network or the governance token, since they are going to be
+            // handled in a special way during the burn itself
+            if (token != _networkToken && token != _govToken) {
+                path[1] = address(networkTokenConverterAnchor(token, registry));
+                path[2] = address(_networkToken);
+            }
+
+            paths[i] = path;
+
+            // make sure to retrieve the balance of either an ERC20 or an ETH reserve
+            if (token == NATIVE_TOKEN_ADDRESS) {
+                amounts[i] = address(feeWallet).balance;
+            } else {
+                amounts[i] = token.balanceOf(address(feeWallet));
+            }
+        }
```

Comment:
This piece of code appears in several other contracts; consider an internal function in Utils.

Reply:
`Utils` shouldn't handle token-specific stuff. I've already opened a task for this.